### PR TITLE
Grid layout for the debugger

### PR DESCRIPTION
**Pull request by yann300**
fix https://github.com/ethereum/remix-project/issues/2712

### CODE REVIEW

File: apps/debugger/src/app/debugger.ts [hunk 1]
```diff
@@ -26,5 +24,7 @@ export class DebuggerClientApi extends DebuggerApiMixin(PluginClient) {
   setFile: (path: string, content: string) => Promise<void>
   getDebugWeb3: () => any // returns an instance of web3.js, if applicable (mainet, goerli, ...) it returns a reference to a node from devops (so we are sure debug endpoint is available)
   web3: () => any // returns an instance of web3.js
+  onStartDebugging: () => void // called when debug starts
+  onStopDebugging: () => void // called when debug starts
```
Comment (Aniket-Engg) [inline apps/debugger/src/app/debugger.ts:28]: typo in comment

File: apps/remix-ide/src/app/panels/layout.ts [hunk 2]
```diff
@@ -64,6 +66,14 @@ export class Layout extends Plugin {
           break
       }
     })
+    this.on('sidePanel', 'focusChanged', async (name) => {
+      const current = await this.call('sidePanel', 'currentFocus')
+      if (this.maximised[current]) {
+        this.event.emit('maximisesidepanel')
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/panels/layout.ts:72]: Not sure if event name should camelcase

Comment (yann300) [reply apps/remix-ide/src/app/panels/layout.ts:72]: the other related event 'minimizesidepanel' was already using this case... I kept the same logic.

**Comment by Aniket-Engg** [comment]
implemented by https://github.com/ethereum/remix-project/pull/2781

**Pull request closed without merging**